### PR TITLE
Indicate usage of the re2 regex engine in the .airflowignore documentation.

### PR DESCRIPTION
**Pull request by tshirtman**
As the engine was changed in #32303, and the feature set of this engine is a bit specific, it seems useful to be explicit about it.

Added a short explanation about it in the dags documenatiton, with a link to the re2 syntax wiki page.

related: #32303

### CODE REVIEW

File: docs/apache-airflow/core-concepts/dags.rst [hunk 1]
```diff
@@ -802,6 +802,10 @@ match any of the patterns would be ignored (under the hood, ``Pattern.search()``
 to match the pattern). Use the ``#`` character to indicate a comment; all characters
 on a line following a ``#`` will be ignored.
 
+As with most regexp matching in Airflow, the regexp engine is `re2`, which explicitly
+doesn’t support many advanced features, please check its
+[documentation](https://github.com/google/re2/wiki/Syntax) for more information.
```
Comment (tshirtman) [inline docs/apache-airflow/core-concepts/dags.rst]: Should have used rst syntax here.

```suggestion
`documentation <https://github.com/google/re2/wiki/Syntax>`_ for more information.
```

**Review by potiuk — APPROVED** [review]

File: docs/apache-airflow/core-concepts/dags.rst [hunk 2]
```diff
@@ -802,6 +802,10 @@ match any of the patterns would be ignored (under the hood, ``Pattern.search()``
 to match the pattern). Use the ``#`` character to indicate a comment; all characters
 on a line following a ``#`` will be ignored.
 
+As with most regexp matching in Airflow, the regexp engine is `re2`, which explicitly
+doesn’t support many advanced features, please check its
```
Comment (tshirtman) [inline docs/apache-airflow/core-concepts/dags.rst]:
```suggestion
doesn't support many advanced features, please check its
```

**Comment by potiuk** [comment]
You need to use double backticks in .rst :)

**Review by pierrejeambrun — APPROVED** [review]

File: docs/apache-airflow/core-concepts/dags.rst [hunk 3]
```diff
@@ -802,6 +802,10 @@ match any of the patterns would be ignored (under the hood, ``Pattern.search()``
 to match the pattern). Use the ``#`` character to indicate a comment; all characters
 on a line following a ``#`` will be ignored.
 
+As with most regexp matching in Airflow, the regexp engine is `re2`, which explicitly
```
Comment (uranusjr) [inline docs/apache-airflow/core-concepts/dags.rst]:
```suggestion
As with most regexp matching in Airflow, the regexp engine is ``re2``, which explicitly
```

**Review by eladkal — APPROVED** [review]

**Pull request merged**